tools/discord/bot.py

```diff
@@ -0,0 +1,246 @@
+import asyncio
+import discord
+from dotenv import load_dotenv
+import os
+from typing import Optional
+
+# Global variable management
+load_dotenv()
+TOKEN = os.getenv("token")
+DOLPHIN_PATH = os.getenv("dolphin_path")
+SP_PATH = os.getenv("sp_path")
+NAND_PATH = os.getenv("nand_path")
+IS_GENERATING_KRKG = False
+
+client = discord.Client(intents=discord.Intents.default())
+tree = discord.app_commands.CommandTree(client)
+
+# ================================
+#     DOLPHIN
+# ================================
+
+
+async def dolphin_clear_io():
+    global NAND_PATH
+
+    rkg_path = os.path.join(NAND_PATH, "test.rkg")
+    krkg_path = os.path.join(NAND_PATH, "test.krkg")
+    ok_path = os.path.join(NAND_PATH, "ok")
+    fail_path = os.path.join(NAND_PATH, "fail")
+
+    if os.path.exists(rkg_path):
+        os.remove(rkg_path)
+
+    if os.path.exists(krkg_path):
+        os.remove(krkg_path)
+
+    if os.path.exists(ok_path):
+        os.remove(ok_path)
+
+    if os.path.exists(fail_path):
+        os.remove(fail_path)
+
+
+async def dolphin_set_ghost(ghost: bytes):
+    global NAND_PATH
+
+    with open(os.path.join(NAND_PATH, "test.rkg"), "wb") as f:
```

Comment:
What happens if `NAND_PATH` env variable is not set? Can we add some kind of check for these global variables? I think just checking that they are non-empty would be fine.

Reply:
The configuration is malformed.

Reply:
Added.